llvm/lib/Transforms/Utils/CodeExtractor.cpp

```diff
@@ -1895,16 +1911,6 @@ CodeExtractor::extractCodeRegion(const CodeExtractorAnalysisCache &CEAC,
 
   fixupDebugInfoPostExtraction(*oldFunction, *newFunction, *TheCall);
 
-  // Mark the new function `noreturn` if applicable. Terminators which resume
-  // exception propagation are treated as returning instructions. This is to
-  // avoid inserting traps after calls to outlined functions which unwind.
-  bool doesNotReturn = none_of(*newFunction, [](const BasicBlock &BB) {
-    const Instruction *Term = BB.getTerminator();
-    return isa<ReturnInst>(Term) || isa<ResumeInst>(Term);
-  });
-  if (doesNotReturn)
-    newFunction->setDoesNotReturn();
-
   LLVM_DEBUG(if (verifyFunction(*newFunction, &errs())) {
```

Comment:
Don't see why this does this, the regular post-pass verification would catch this?

Reply:
Tested a simple case w.o this (using `HotColdSplitting`). If we don't add `noreturn` here, it never gets added in defaultO3 pipeline.

Reply:
NB the comment around when `HotColdSplitting` is enabled:
```
  // Split out cold code. Splitting is done late to avoid hiding context from
  // other optimizations and inadvertently regressing performance. The tradeoff
  // is that this has a higher code size cost than splitting early.
```

Reply:
I wasn't talking about anything the pass does, I mean the verifyFunction call

Reply:
Oh, not sure what the need for that is.